include/zephyr/bluetooth/mesh/health_srv.h

```diff
@@ -175,9 +175,9 @@ struct bt_mesh_health_srv {
  *  @return New mesh model instance.
  */
 #ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV
-#define BT_MESH_MODEL_HEALTH_SRV(srv, pub)                                              \
-	BT_MESH_MODEL_METADATA_CB(BT_MESH_MODEL_ID_HEALTH_SRV, bt_mesh_health_srv_op,   \
-			 pub, srv, &bt_mesh_health_srv_cb, &(srv)->metadata)
+#define BT_MESH_MODEL_HEALTH_SRV(srv, pub, meta)                                      \
+	BT_MESH_MODEL_METADATA_CB(BT_MESH_MODEL_ID_HEALTH_SRV, bt_mesh_health_srv_op, \
+			 pub, srv, &bt_mesh_health_srv_cb, meta)
```

Comment:
What if we try to get rid of `#ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` - `#else` - `#endif` in every model header?

Suggestion 1: Define a dedicated macro for metadata use case. It can even be used without Large Composition Data Server as access.h handles this case in `BT_MESH_MODEL_METADATA_CB` by ignoring the metadata argument if LCD srv is disabled: https://github.com/zephyrproject-rtos/zephyr/blob/b9f3d68863b9f75dcc1dfac7dd7ff269b99cd118/include/zephyr/bluetooth/mesh/access.h#L511-L529

Example: https://github.com/zephyrproject-rtos/zephyr/commit/beca53b3534f0a47c13647d0f2851ed11ebe4c61

----
Suggestion 2: Use a single macro for both cases (like in your change) but without `#ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` - `#else` - `#endif` in every model header (change applies to the commit above): https://github.com/zephyrproject-rtos/zephyr/commit/a513297d99f66440b4d6e98a0230b58607c8acc5

Branch for testing: https://github.com/PavelVPV/zephyr/tree/metadata_without_extra_pointer

Reply:
What may not be good in these suggestions is that if LCD srv is disabled, a code will still compile without warnings. If it was done accidentally, the code will give impression that it includes metadata while it isn't.

In the second suggestion I'm not sure how doxygen will handle `@param meta` without actual parameter in the macro.

Reply:
I like the second suggestion but with two changes:
1. It should be just `__VA_ARGS__` instead of `GET_ARG_N(1, __VA_ARGS__)`. If the user provides some strange combination of parameters or mistyping, it will be caught on compilation stage.
2. It should be `@param ...` instead of `@param meta`. Doxygen recognizes `...` syntax, I checked it. Example: https://github.com/nrfconnect/sdk-nrf/pull/14261/files#diff-611e9a4b1b67a364f2a66fcb3a7555dc20625487d8894abcbb77c403c607acddR86

Reply:
I guess impression doesn't matter about metadata is in or isn't. Specification strictly claims when it is available and used. Metadata is cut out because there is no place to store it (no responsible composition data) according to specification.

Reply:
1. Ok
2. Yeah, but how do customers know that the 3rd argument is metadata?

Reply:
2. In description of `@param ...`

Reply:
Ok. Option2 with __VA_ARGS__ looks fine. I will update accordingly and check compilation in both cases.

Reply:
Actually, I overcomplicated the second variant. It can be just (https://github.com/PavelVPV/zephyr/commit/514351d02192171895c6d2db359a6e6bf14dea0b):
```
 #define BT_MESH_MODEL_HEALTH_SRV(srv, pub, ...)                                \
       BT_MESH_MODEL_METADATA_CB(BT_MESH_MODEL_ID_HEALTH_SRV, bt_mesh_health_srv_op, \
                                 pub, srv, &bt_mesh_health_srv_cb, \
                                 __VA_ARGS__)
```

Reply:
Please check new commit.

Reply:
I'm not sure it is good to change `BT_MESH_MODEL_METADATA_CB` to use variable arguments as it is specifically added for models with metadata.